src/soil_plant_hydrology_model.jl

```diff
@@ -55,8 +55,8 @@ function SoilPlantHydrologyModel{FT}(;
 }
 
     #These may be passed in, or set, depending on use scenario
-    top_flux_bc = FluxBC(FT(0.0))
-    bot_flux_bc = FluxBC(FT(0.0))
+    top_flux_bc = FluxBC{FT}((p, t) -> eltype(t)(0.0))
```

Comment:
should this be `FT(0.0)` instead of `eltype(t)(0.0)`?

Reply:
Script to test:
```module Tmpm
export FluxBC, f

struct FluxBC{T}
    bc::Function
end

function f(bc::FluxBC{T}, t::T) where {T}
    for i in 1:1000
        bc.bc(t)
    end
end

end
```

```
include("Tmpm.jl")
using .Tmpm

using BenchmarkTools
FT = Float32
bc1 = FluxBC{FT}((t) -> FT(0.0))
bc2 = FluxBC{FT}((t) -> eltype(t)(0))
f(bc1, FT(0.0));
f(bc2, FT(0.0));
@btime f(bc1,FT(0))
@btime f(bc2,FT(0))
```

results:
```
julia> include("./tmp.jl")
  89.417 μs (2001 allocations: 31.27 KiB)
  9.375 μs (1001 allocations: 15.64 KiB)
```

so we should be using "eltype(t)" rather than "FT"!


Im not sure we need the `FT` in the BC structs, then. it's never being used.

Reply:
I removed `FT` from all the BC structs, tests still pass! we can add it in to specific instances as needed.